New API file for Gifts and gift Items.

maasj opened the pull request:
These API endpoints are useful for importing donations from a legacy system.

gbeeley commented on kardia-app/modules/rcpt/api_gift.qyt:
```diff
@@ -0,0 +1,111 @@
+                				    where
+                                        :g:a_ledger_number = :ledgers_list:a_ledger_number and
+                                        :g:a_batch_number = :ledgerbatches_list:a_batch_number
+                				    group by
```
The `GROUP BY` clause is not needed in this query since it is just specifying the unique primary key and so won't result in any change in the result set.  You could replace it by an `ORDER BY :g:a_gift_number`.

gbeeley commented on kardia-app/modules/rcpt/api_gift.qyt:
```diff
@@ -0,0 +1,111 @@
+                                                :i:a_ledger_number = :ledgers_list:a_ledger_number and
+                                                :i:a_batch_number = :ledgerbatches_list:a_batch_number and
+                                                :i:a_gift_number = :ledgerbatchgifts_list:a_gift_number
+                                            group by
```
Similar to above comment - `GROUP BY` is not needed here.

gbeeley reviewed (changes requested):
Just a couple of minor changes needed.  Thanks!

maasj commented:
I removed the GROUP BY clauses and added what I think are the proper ORDER BY clauses.
I've also since added an endpoint for basic interactions with the electronic giving import table.  Hopefully that's OK.  😀
Let me know if more changes are needed.

gbeeley commented on kardia-app/modules/rcpt/api_gift.qyt:
```diff
@@ -0,0 +1,123 @@
+                                                :i:a_batch_number = :ledgerbatches_list:a_batch_number and
+                                                :i:a_gift_number = :ledgerbatchgifts_list:a_gift_number
+                                            order by
+                                                :i:a_gift_number,
```
`a_gift_number` not needed in the `ORDER BY` clause, since it's uniquely specified in the `WHERE` clause.

gbeeley reviewed (changes requested):
One minor change needed.  Thanks!

maasj commented:
Fixed!

gbeeley reviewed (approved):
Thanks massj!
